github/VisualStudio
# Hide resolved comments from changes tree and indicate resolved comments on margin

Previously pull request comments would be implicitly resolved when when the text near to the comment was changed. GitHub now supports comments being explicitly resolved by a user _and_ comments becoming implicitly outdated when the text near to the comment is changed.

This pull request surfaces the `IsResolved` status of a review comment threads and hides resolved comments from the changes tree (in a similar way to how outdated comments are hidden). This makes it easy to find comments that still need to be addressed when responding to a pull request review.

### What this PR does

- Use `pullRequest/reviewThreads` to populate the `IsResolved` status of a pull request comment threads
- When using GitHub Enterprise fall back to using original model (`reviewThreads` might not be available)
- Hide resolved comments from the changes tree
- Show different in-line comment glyph for resolved comments (an empty circle)
- Show banner to clearly indicate when a thread has been resolved

### To do

- [x] Support fallback for older versions of GitHub Enterprise
- [x] Indicate outdated in-line comments in glyph
![image](https://user-images.githubusercontent.com/11719160/53804701-1e47f580-3f40-11e9-84fa-fb0ea3764dea.png)
- [x] Show banner on resolved comment threads
![image](https://user-images.githubusercontent.com/11719160/53804870-88f93100-3f40-11e9-94cc-2a249839b46a.png)

### What this PR doesn't do

- Probe GitHub Enterprise schema to see if we could fetch resolved comment information
- Surface comments that are outdated but unresolved
- Allow resolving of comments using inline comment view

### How to test

1. Open `https://github.com/grokys/PullRequestSandbox` in Visual Studio
2. Open grokys/PullRequestSandbox#58 in `GitHub` pane
3. Check that only 1 comment appears on `Program.cs`
![image](https://user-images.githubusercontent.com/11719160/53812034-8273b500-3f52-11e9-9024-855349fbe187.png)
4. Click on `💬 1` comment
5. Check `Program.cs` appears in diff view and comment is visible
6. Check that empty circle glyph appears below indicating a resolved comment
7. Click on the resolved comment glyph
8. Check that `The conversation was marked as resolved` banner appears above comment

### Notes

Here is what is looks like in the `Dark` theme

![image](https://user-images.githubusercontent.com/11719160/53812533-9b309a80-3f53-11e9-8064-26bc49685d7b.png)

Related #2245
